File: go/internal/booking-service/core/application/guest.manager.go
package application

import (
	"book/internal/booking-service/adapters/repositories"
	"book/internal/booking-service/core/application/guest/requests"
	"book/internal/booking-service/core/application/guest/responses"
	sharedResponse "book/internal/shared/application/responses"
)

type GuestManager struct {
	guestRepository *repositories.GuestRepository
}

func NewGuestManager(guestRepository *repositories.GuestRepository) *GuestManager {
	return &GuestManager{
		guestRepository: guestRepository,
	}
}

func (gm GuestManager) CreateGuest(request requests.GuestRequest) responses.GuestResponse {
	guest := request.Data.MapToEntity()
	err := gm.guestRepository.Create(guest)

	if err != nil {
		return responses.GuestResponse{
			Response: sharedResponse.Response{
				Success:   false,
				Message:   "There was an error when saving to DB",
				ErrorCode: sharedResponse.COULDNOT_STORE_DATA,
			},
		}
	}

	return responses.GuestResponse{
		Response: sharedResponse.Response{
			Success: true,
		},
	}
}

File: nestjs/src/booking/core/application/guest.manager.ts
import { AppResponse } from "@/shared/application/responses/response";
import { IGuestRepository } from "../domain/ports/guest.repository";
import { IGuestManager } from "./guest/ports/iguest-manager.port";
import { GuestRequest } from "./guest/requests/guest.request";
import { GuestResponse } from "./guest/responses/guest.response";

export class GuestManager implements IGuestManager {
    constructor(private readonly guestRepository: IGuestRepository){}

    async Create(request: GuestRequest): Promise<GuestResponse> {
        try {
            const guest = GuestRequest.mapToEntity(request.data);
            request.data.id = await this.guestRepository.Create(guest);

            return {
                data: request.data,
                success: true,
                message: ""
            };
        } catch(err) {
            return {
                success: false,
                message: "There was an error when saving to DB",
                errorCode: AppResponse.ErrorCodes.COULDNOT_STORE_DATA
            }
        }
    }
}